go/vt/vtgateproxy/discovery.go

```diff
@@ -319,11 +325,11 @@ func (b *JSONGateResolverBuilder) update(r *JSONGateResolver) {
 		}
 	}
 	if unknown != 0 {
-		affinityCount.Add("unknown", unknown)
+		affinityCount.Set("unknown", unknown)
 	}
-	affinityCount.Add("local", local)
-	affinityCount.Add("remote", remote)
-	poolTypeCount.Add(r.poolType, int64(len(targets)))
+	affinityCount.Set("local", local)
```

Comment:
As-is this isn't quite accurate, since the set is already filtered by target by this point. We could move the 

I realize counters aren't quiite right either, but using gauges across the fleet in prometheus isn't great for aggregations.

Reply:
One thing we could do is pivot the model so that instead of counting whenever we update a target, we could instead use something like `stats.NewGaugesFuncWithMultiLabels` so that we generate the stats on demand regardless of whether or not any clients are connected.

I think that would be a better model frankly, since it would exercise all the parsing code and we could verify the discovery layer is running as we expect before sending any traffic through the system.

We could also wire in a /debug/status page so we can see the full state as well.